File: backend/app/api/v1/ai.py
from fastapi import APIRouter, Depends

from app.api.deps import CurrentUser, require
from app.core.ai import get_ai_provider
from app.core.rbac import Permission
from app.schemas.ai import ControlSuggestionOut, SuggestControlsRequest

router = APIRouter(prefix="/ai", tags=["ai"])


@router.post("/suggest-controls", response_model=list[ControlSuggestionOut])
async def suggest_controls(
    body: SuggestControlsRequest,
    _user: CurrentUser = Depends(require(Permission.CONTROL_CREATE)),
):
    """AI control suggestions for a risk (Phase 6). Provider is pluggable (Q4)."""
    provider = get_ai_provider()
    suggestions = await provider.suggest_controls(
        risk_name=body.risk_name, risk_description=body.risk_description
    )
    return [
        ControlSuggestionOut(
            name=s.name, description=s.description, control_type=s.control_type
        )
        for s in suggestions
    ]

File: frontend/src/api/ai.ts
import { api } from "./client";

export interface AiStatus {
  provider: string;
  active: string;
  model: string | null;
}
export interface ControlSuggestion {
  name: string;
  description: string | null;
  control_type: string | null;
}

export const aiStatus = () => api<AiStatus>(`/ai/status`);

export const suggestControls = (risk_name: string, risk_description?: string | null) =>
  api<ControlSuggestion[]>(`/ai/suggest-controls`, {
    method: "POST",
    body: { risk_name, risk_description: risk_description ?? null },
  });
